client/components/account-balances/balances-tab-panel.tsx

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { useAllDepositsOverviews } from 'wcpay/data';
+import { Flex, TabPanel } from '@wordpress/components';
+
+/**
+ * Internal dependencies.
+ */
+import { fundLabelStrings } from './strings';
+import { getCurrencyTabTitle } from './utils';
+import BalanceBlock from './balance-block';
+
+/**
+ * Renders an account balances panel with tab navigation for each deposit currency.
+ *
+ * @return {React.FunctionComponent} Rendered balances panel with tab navigation for each currency.
+ */
+const AccountBalancesTabPanel: React.FC = () => {
+	const {
+		overviews,
+		isLoading,
+	} = useAllDepositsOverviews() as AccountOverview.OverviewsResponse;
+
+	// While the data is loading, we show the default currency tab.
+	let depositCurrencyTabs: TabPanel.Tab[] = [
+		{
+			name: wcpaySettings.accountDefaultCurrency,
+			title: getCurrencyTabTitle( wcpaySettings.accountDefaultCurrency ),
+		},
+	];
+
+	const { currencies } = overviews;
+
+	if ( ! isLoading && currencies.length !== 0 ) {
+		depositCurrencyTabs = currencies.map(
+			( overview: AccountOverview.Overview ) => ( {
+				name: overview.currency,
+				title: getCurrencyTabTitle( overview.currency ),
+				currencyCode: overview.currency,
+				availableFunds: overview.available.amount,
+				pendingFunds: overview.pending.amount,
+				reserveFunds: 0, // TODO: Add reserve funds to the overview object.
+			} )
+		);
+	}
+
+	return (
+		<TabPanel tabs={ depositCurrencyTabs }>
+			{ ( tab ) => (
+				<Flex gap={ 0 } className="wcpay-account-balances__balances">
+					<BalanceBlock
+						title={ fundLabelStrings.available }
+						amount={ tab.availableFunds }
+						currencyCode={ tab.currency_code }
+						isLoading={ isLoading }
+					/>
+					<BalanceBlock
+						title={ fundLabelStrings.pending }
+						amount={ tab.pendingFunds }
+						currencyCode={ tab.currency_code }
```

Comment:
When using the currency code `JPY` it is incorrectly formatted and displayed as `$`.

<img width="702" alt="image" src="https://user-images.githubusercontent.com/3147296/225503689-8e49a0bf-5a88-48f5-b7ba-91524b4a05e7.png">


This is because `currencyCode` is being passed as `undefined` to `<BalanceBlock />` when accessing `currency_code` by mistake.

This is easy to miss since the `TabPanel.Tab` type is allowing any combo of keys/values, so we're missing the opportunity to declare `currencyCode` as a required property.

I recommend creating a `BalanceTab` type and expecting it when mapping over the tabs:

```ts
type BalanceTab = {
	name: string;
	title: string;
	currencyCode: string;
	availableFunds: number;
	pendingFunds: number;
	reservedFunds: number;
};
```
```tsx
		<TabPanel tabs={ depositCurrencyTabs }>
			{ ( tab: BalanceTab ) => (
```
```tsx
let depositCurrencyTabs: BalanceTab[] = [
```

Reply:
Nice catch and I think this issue goes to show how good TS can be when you use type and know what you're doing. 😅 

I've had a crack at your suggested changes in this commit. Let me know if I missed anything. [e26e0a6](https://github.com/Automattic/woocommerce-payments/pull/5764/commits/e26e0a684f75784c2a8bc03f896a4d3bd513ae50)

Reply:
Yeh looks good!

It took me ages to realise that this was caused by `currency_code` since the human eye doesn't easily notice the difference between `currencyCode`/`currency_code`.

But a red squiggly 👌

<img width="539" alt="image" src="https://user-images.githubusercontent.com/3147296/225509723-430f0314-1ace-4b47-bda0-8e7e9ac3291b.png">